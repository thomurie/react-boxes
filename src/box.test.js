import {
  queryByAltText,
  queryByTestId,
  render,
  screen,
} from "@testing-library/react";
import Box from "./box";

// Smoke
test("renders learn react link", () => {
  render(<Box />);
});

// Snapshot
test("Matches snapshot", () => {
  const { asFragment } = render(<Box />);
  expect(asFragment()).toMatchSnapshot();
});

// Critical Business Logic
test("renders box in document", () => {
  const { queryByTestId } = render(
    <Box width={200} height={200} color="yellow" />
  );
  expect(queryByTestId("yellow")).toBeInTheDocument();
});

test("renders remove button in document", () => {
  render(<Box width={200} height={200} color="yellow" />);
  expect(screen.queryByText(/x/i)).toBeInTheDocument();
});
