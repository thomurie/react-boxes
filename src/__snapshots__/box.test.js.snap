// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Box"
    data-testid="orange"
    style="background-color: orange; width: 200px; height: 200px;"
  >
    <button
      class="Box"
    >
      X
    </button>
  </div>
</DocumentFragment>
`;
